dicodile/workers/dicod_worker.py

```diff
@@ -668,7 +671,11 @@ def recv_D(self):
         """Receive a dictionary D"""
         if self._backend == "mpi":
             comm = MPI.Comm.Get_parent()
-            self.D = recv_broadcasted_array(comm)
+            if self.rank1:
+                self.uv = recv_broadcasted_array(comm)
```

Comment:
does this works? not sure this function is meant to be used with a tuple.

Reply:
Good point, definitely not working yet